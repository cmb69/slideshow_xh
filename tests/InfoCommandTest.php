<?php

/**
 * Copyright 2021 Christoph M. Becker
 *
 * This file is part of Slideshow_XH.
 *
 * Slideshow_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Slideshow_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Slideshow_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Slideshow;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\View;

class InfoCommandTest extends TestCase
{
    public function testRendersView()
    {
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["slideshow"]);
        $sut = new InfoCommand("./plugins/slideshow/", $view, new FakeSystemChecker());
        $response = $sut();
        $this->assertSame("Slideshow 1.4-dev", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
