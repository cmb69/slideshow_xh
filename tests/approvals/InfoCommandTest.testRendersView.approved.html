
<!-- Slideshow_XH info -->
<h1>Slideshow 1.4-dev</h1>
<h2>Synopsis</h2>
<pre>&lt;?=slideshow('FOLDER', 'OPTIONS')?&gt;
{{{slideshow('FOLDER', 'OPTIONS')}}}</pre>
<dl>
  <dt>FOLDER:</dt>
  <dd>The path of a folder relative to CMSimple_XH's image folder.</dd>
  <dt>OPTIONS:</dt>
  <dd>Optional settings in form of a query string (option1=value1&amp;option2=value2). The following options are available:  <dl>
    <dt>order</dt>
    <dd>The order of the images: &quot;fixed&quot; (alphabetically ordered; start with first image), &quot;sorted&quot; (alphabetically sorted; start with randomly chosen image) or &quot;random&quot; (randomly ordered).</dd>
    <dt>effect</dt>
    <dd>The kind of the transition.</dd>
    <dt>easing</dt>
    <dd>The acceleration effect.</dd>
    <dt>delay</dt>
    <dd>The initial delay in ms until the slideshow starts.</dd>
    <dt>pause</dt>
    <dd>The duration of the pause between the transitions in ms.</dd>
    <dt>duration</dt>
    <dd>The duration of the transition effect in ms.</dd>
  </dl></dd>
</dl>
<h2>System check</h2>
<p class="xh_fail">PHP version ≥ 7.1.0</p>
<p class="xh_fail">CMSimple_XH version ≥ 1.7.0</p>
<p class="xh_fail">Plib_XH version ≥ 1.2</p>
<p class="xh_warning">Folder './plugins/slideshow/config/' writable</p>
<p class="xh_warning">Folder './plugins/slideshow/css/' writable</p>
<p class="xh_warning">Folder './plugins/slideshow/languages/' writable</p>
